### PR TITLE
Feature/1176142788936875/trash bin

### PR DESCRIPTION
**Pull request by Alfredo-Moreira**
## PR Checklist

- [ ] Assign @FJKhan **and** @Alfredo-Moreira as reviewers.
- [ ] If your PR is not a hotfix, is it targeted for `dev`? If your PR is a hotfix, is it targeted for `master`?
- [ ] Unit and functional test coverage was added where applicable.
- [ ] CI/CD passes for your PR.
- [ ] Complex code is well documented with comments.
- [ ] Does the original ticket have test instructions? If not add them below

### CODE REVIEW

File: cypress/integration/dashboard_spec.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //Test Suite
 describe('Home Page Dashboard Tests', () => {
 
-    let viewports = [Cypress.env('desktop'),Cypress.env('tablet'),Cypress.env('mobile')];
+    let viewports = [Cypress.env('desktop')];
```
Comment (FJKhan) [inline cypress/integration/dashboard_spec.js:10]: are these changes deliberate?

File: cypress/support/reusable_tests/admin.js [hunk 2]
```diff
@@ -87,31 +88,40 @@ Cypress.Commands.add('testAdminPageElements',(viewport,creds)=>{
 Cypress.Commands.add('testAdminFilterUsers',(viewport,creds)=>{
     cy.viewport(viewport);
     cy.login(creds.email,creds.password);
-
+    var type = 'lawyer';
```
Comment (FJKhan) [inline cypress/support/reusable_tests/admin.js:91]: any particular reason we're using the lawyer user type here?

File: cypress/support/reusable_tests/admin.js [hunk 3]
```diff
@@ -193,7 +203,209 @@ Cypress.Commands.add('testAdminFilterAddNewManagerAction',(viewport,creds,admin)
...
+     cy.intercept('/v1/organizations/**');
+
+    
+    //Intercept
+    cy.intercept('/v1/organizations?deleted=true').as('deletedOrganizations');
+    cy.intercept('/v1/organizations?serviceDeleted=true').as('deletedOrgServices');
+    //Click
+    cy.getElementByTestId('header-admin-link').click();
+     
+    cy.wait(['@deletedOrganizations','@deletedOrgServices'])
```
Comment (FJKhan) [inline cypress/support/reusable_tests/admin.js:223]: I think this wait is in the wrong place. it should go after said actions are invoked e.g. whatever action triggers the network request

File: cypress/support/reusable_tests/admin.js [hunk 4]
```diff
@@ -193,7 +203,209 @@ Cypress.Commands.add('testAdminFilterAddNewManagerAction',(viewport,creds,admin)
...
+    
+    cy.getElementByTestId('admin-tab-trash-bin').click();
+
+    cy.get('@deletedOrgServices').then(interception =>{
+        if(interception.response.body.organizations.length){
+            if(interception.response.body.organizations.length){
```
Comment (FJKhan) [inline cypress/support/reusable_tests/admin.js:389]: duplicate if statement

**Comment by FJKhan** [comment]
1. can we change "View Deleted Service Organization" to just "View organization" ? it's lengthy & confusing

image

2. catalog is still showing deleted service. we should update the api to filter out the deleted orgs/services as well
image

3. I think we should include the service name in the restore service dialog 
image

File: cypress/integration/dashboard_spec.js (hunk 1 above)
Comment (Alfredo-Moreira) [reply cypress/integration/dashboard_spec.js:10]: Yes, So far from what i have seen there are no code changes between Desktop, tablet and mobile. The app just resizes to fit the screen. It was running the same test 3 times. I can however bring them back.

File: cypress/support/reusable_tests/admin.js (hunk 2 above)
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js:91]: Just one of the options available that is not a regular user.

File: cypress/support/reusable_tests/admin.js (hunk 3 above)
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js:223]: The cy.login triggers a response. Immediately after the login, the redirect to dashboard triggers the organizations endpoint.

File: cypress/support/reusable_tests/admin.js [hunk 5]
```diff
@@ -193,7 +203,209 @@ Cypress.Commands.add('testAdminFilterAddNewManagerAction',(viewport,creds,admin)
...
+    cy.intercept('/v1/organizations?serviceDeleted=true').as('deletedOrgServices');
+    //Click
+    cy.getElementByTestId('header-admin-link').click();
+     
+    cy.wait(['@deletedOrganizations','@deletedOrgServices'])
+    
+    cy.getElementByTestId('admin-tab-trash-bin').click();
+
+    cy.get('@deletedOrganizations').then(interception =>{
+        if(!interception.response.body.organizations.length){
```
[the remark this answers is not in the document]
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js:228]: I can update this.

File: cypress/support/reusable_tests/admin.js (hunk 4 above)
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js:389]: nice catch!

File: src/components/AdminPanelTrashBin.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+import Table from '../components/Table';
+import {Container, SectionTitle, Title} from '../components/styles';
+import {CATALOG_API_URL} from '../utils';
+import {useAPIGet} from '../utils/hooks';
+
+const AdminPanelTrashbin = (props) => {
+    const {closeModal, openModal} = useContext(ContextFormModal);
+    const deletedOrgs = useAPIGet(`/organizations?deleted=true`)?.data?.organizations;
+    const deletedServices = useAPIGet(`/organizations?serviceDeleted=true`);
+    const reducedDeletedServices = deletedServices?.data?.organizations?.reduce(
```
[the remark this answers is not in the document]
Comment (Alfredo-Moreira) [reply src/components/AdminPanelTrashBin.js:16]: Yes, by default the organizations api (/organizations) returns organizations that are not deleted, and without deleted services. By using the query param, I can force that case scenario and populate the trash bin.

[the remark this answers is not in the document]
Comment (Alfredo-Moreira) [reply src/components/AdminPanelTrashBin.js:16]: But i will take a look at that. I think I might have had that there before the changes I implemented in the API

**Review by Alfredo-Moreira — COMMENTED** [review]
Saw all your comments. I will begin addressing them.

File: cypress/support/reusable_tests/admin.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js:228]: In order to address this, I will have to create a function to clear deleted orgs/services and set them to is_deleted: false to test the empty state, and then and org that is deleted and has deleted services.

[the remark this answers is not in the document]
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js]: https://github.com/asylum-connect/control-panel/pull/69/commits/9c12ec4b1128256d673c392446de1fbc37c5533a

File: cypress/support/reusable_tests/admin.js (hunk 3 above)
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js]: Admin Link- routes to endpoint( /admin ), this endpoint does all the calls at once for all 3 available tabs(Users,Suggestions,Trash Bin) under that option. So I have to intercept after that click

File: cypress/support/reusable_tests/admin.js (hunk 4 above)
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js]: Removed

File: src/components/AdminPanelTrashBin.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Alfredo-Moreira) [reply src/components/AdminPanelTrashBin.js:16]: Addressed the reduce by reduce the if statement with https://github.com/asylum-connect/control-panel/pull/69/commits/cd38af148e3a772ad2fd336727b3f90d1adc7c08 . But I use the reduce to make the array with services

[the remark this answers is not in the document]
Comment (Alfredo-Moreira) [reply src/components/AdminPanelTrashBin.js:16]: The API currently only addresses that with [v1/organizations/:id](https://github.com/asylum-connect/catalog-api/blob/dev/src/routes/organizations.js#L209) query. Unfortunately, the [v1/organizations?query](https://github.com/asylum-connect/catalog-api/blob/dev/src/routes/organizations.js#L19) type of endpoint is a straight query and not an aggregate pipeline. This is why I'm using the reduce to filter out the services that are not deleted. That query only returns all the orgs that have deleted services. Turning that into an aggregate would an huge undertaking, as the call handles a lot of other query params.

**Comment by Alfredo-Moreira** [comment]
> 

Addressed:
 1. Addressed it by removing extra text
 2. Looking into this
 3. Had to add message display support on the formModal. Not in body of modal it will display the name of the organization or service being deleted or restored.

File: cypress/support/reusable_tests/admin.js (hunk 2 above)
Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js:90]: Do you want me to test with a random selection of the options? I can do that and not test with a static selection.

Comment (Alfredo-Moreira) [reply cypress/support/reusable_tests/admin.js]: I added all possible types and tested it

**Review by FJKhan — APPROVED** [review]
